src/wombats/game/finalizers.clj

```diff
@@ -11,19 +11,43 @@
                             [] meta)]
     (assoc cell :meta meta-update)))
 
-(defn- update-metadata-decay
-  [game-state]
+(defn- update-deterioration
+  "Determines the deterioration level and applys it to the element"
+  [{:keys [contents] :as cell} max-hp]
+  (let [percent-deteriorated (-> (:hp contents)
+                                 (/ max-hp)
+                                 (float))
```

Comment:
What does casting it as a float after division do? Wouldn't we need to cast it before the division so we ensure non int division?